### PR TITLE
Iiwa 190

### PR DESCRIPTION
**Pull request by mcdowellalex**

### CODE REVIEW

File: src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx [hunk 1]
```diff
@@ -191,6 +191,7 @@ export const DashboardPage: React.FC<IDashboardPageProps> = props => {
         axios
             .get('/dashboard/CostPieCharts')
             .then((data: any) => {
+                console.log(data)
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx]: remove

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx [hunk 2]
```diff
@@ -362,40 +420,42 @@ export const HardwareDetailEditPage: React.SFC<IHardwareDetailEditPageProps> = p
...
-                            : null,
-                        MonthsPerRenewal: hasRecurringCost ? costSection[2] : null,
+                            : 0,
+                        MonthsPerRenewal: hasRecurringCost ? costSection[2] : 0,
 
                         TextField: commentText,
                     },
                 })
             } else if (match.params.type === 'laptop') {
+                console.log(firstSectionData)
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: remove

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx [hunk 3]
```diff
@@ -285,6 +290,69 @@ export const HardwareDetailEditPage: React.SFC<IHardwareDetailEditPageProps> = p
...
+        if (match.params.type === 'server') {
+            if (isNaN(Number(firstSectionData[3]))) {
+                alertMssg += '\n RAM must be a number!'
+            }
+            if (isNaN(Number(firstSectionData[5]))) {
+                alertMssg += '\n # of cores must be a number! '
+            }
+        }
+
+        if (alertMssg.length > 0) {
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: move this

File: src/frontend-admin/src/components/reusables/PieChart/PieChart.tsx [hunk 4]
```diff
@@ -24,6 +24,9 @@ interface IRechartPieProps {
 export const RechartPieChart: React.FunctionComponent<IRechartPieProps> = props => {
     const {pieChartData, initialColors, onSliceClick} = props
 
+    if (pieChartData.length === 0) {
```
Comment (kirbur) [inline src/frontend-admin/src/components/reusables/PieChart/PieChart.tsx]: what is this for?

File: src/frontend-admin/src/components/pages/DepartmentsListPage/DepartmentsListPage.tsx [hunk 5]
```diff
@@ -267,12 +269,16 @@ export const DepartmentsListPage: React.SFC<IDepartmentsListPageProps> = props =
...
-                        <td className={styles.alignLeft}>
+                        <td className={styles.alignLeft} key={i}>
                             {row[2] === 1 ? row[2] + ' employee' : row[2] + ' employees'}
                         </td>
                     )
                 case 3:
-                    transformedRow[3] = <td className={styles.alignLeft}>{formatCost(row[3])}</td>
+                    transformedRow[3] = (
+                        <td className={styles.alignLeft} key={i}>
+                            {formatCost(row[3])}
```
Comment (freakingawesome) [inline src/frontend-admin/src/components/pages/DepartmentsListPage/DepartmentsListPage.tsx]: @mcdowellalex - if you're running out of things to do, there's always the option of refactoring to get away from hardcoding indices everywhere!

**Pull request merged**